code/modules/projectiles/projectile.dm

```diff
@@ -67,10 +67,26 @@
 	qdel(src)
 
 /obj/item/projectile/proc/on_hit(atom/target, blocked = 0, hit_zone)
+	var/turf/target_loca = get_turf(target)
 	if(!isliving(target))
 		return 0
 	var/mob/living/L = target
 	if(blocked < 100) // not completely blocked
+		if(damage && L.blood_volume && damage_type == BRUTE)

```

Comment:
It checks damage here @shazbot194.

If the projectile does brute damage to a mob that has blood, splatter (with a 1/3 chance to actually put blood on the ground)